network/network.go

```diff
@@ -411,10 +412,28 @@ func (n *Network) validateNodeDID(nodeDID did.DID) error {
 	// Check if the DID document has a resolvable NutsComm endpoint
 	serviceResolver := doc.NewServiceResolver(n.didDocumentResolver)
 	serviceRef := doc.MakeServiceReference(nodeDID, transport.NutsCommServiceType)
-	_, err = serviceResolver.Resolve(serviceRef, doc.DefaultMaxServiceReferenceDepth)
+	nutsCommService, err := serviceResolver.Resolve(serviceRef, doc.DefaultMaxServiceReferenceDepth)
 	if err != nil {
 		return fmt.Errorf("unable to resolve %s service endpoint, register it on the DID document (did=%s): %v", transport.NutsCommServiceType, nodeDID, err)
 	}
+
+	// Self-authentication requires a certificate. Running non-TLS is evaluated in Configure
+	if n.certificate.Leaf == nil { // n.config.DisableNodeAuthentication is for incoming connections.
```

Comment:
shouldn't NutsComm still be checked (presence and validated, but not vs certificate)?

Reply:
This should probably return an error, not nil. Strict mode won't get here, and in non-strict mode it doesn't do any harm to have an error logged for missing TLS certificate.

Reply:
moved this further down the checklist and now returns an error